app/command_remove.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	"strings"
+
+	l4g "github.com/alecthomas/log4go"
+	goi18n "github.com/nicksnyder/go-i18n/i18n"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/utils"
+)
+
+type RemoveProvider struct {
+}
+
+type KickProvider struct {
+}
+
+const (
+	CMD_REMOVE = "remove"
+	CMD_KICK   = "kick"
+)
+
+func init() {
+	RegisterCommandProvider(&RemoveProvider{})
+	RegisterCommandProvider(&KickProvider{})
+}
+
+func (me *RemoveProvider) GetTrigger() string {
+	return CMD_REMOVE
+}
+
+func (me *KickProvider) GetTrigger() string {
+	return CMD_KICK
+}
+
+func (me *RemoveProvider) GetCommand(T goi18n.TranslateFunc) *model.Command {
+	return &model.Command{
+		Trigger:          CMD_REMOVE,
+		AutoComplete:     true,
+		AutoCompleteDesc: T("api.command_remove.desc"),
+		AutoCompleteHint: T("api.command_remove.hint"),
+		DisplayName:      T("api.command_remove.name"),
+	}
+}
+
+func (me *KickProvider) GetCommand(T goi18n.TranslateFunc) *model.Command {
+	return &model.Command{
+		Trigger:          CMD_KICK,
+		AutoComplete:     true,
+		AutoCompleteDesc: T("api.command_remove.desc"),
+		AutoCompleteHint: T("api.command_remove.hint"),
+		DisplayName:      T("api.command_kick.name"),
+	}
+}
+
+func (me *RemoveProvider) DoCommand(a *App, args *model.CommandArgs, message string) *model.CommandResponse {
+	return doCommand(a, args, message)
+}
+
+func (me *KickProvider) DoCommand(a *App, args *model.CommandArgs, message string) *model.CommandResponse {
+	return doCommand(a, args, message)
+}
+
+func doCommand(a *App, args *model.CommandArgs, message string) *model.CommandResponse {
+	channel, err := a.GetChannel(args.ChannelId)
+	if err != nil {
+		return &model.CommandResponse{Text: args.T("api.command_channel_rename.channel.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+	}
+
+	if channel.Type == model.CHANNEL_OPEN && !a.SessionHasPermissionToChannel(args.Session, args.ChannelId, model.PERMISSION_MANAGE_PUBLIC_CHANNEL_PROPERTIES) {
```

Comment:
Should this be `model.PERMISSION_MANAGE_PUBLIC_CHANNEL_MEMBERS`?

Reply:
thats true, thanks!! done @ccbrown